File: AgentApp/Frontend/src/app/service/price-list.service.ts
import { Router } from '@angular/router';
import { PriceList } from './../model/priceList';
import { environment } from './../../environments/environment';
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class PriceListService {
  url = environment.baseUrl + environment.priceList;
  createSuccessEmitter = new Subject<PriceList>();

  constructor(private httpClient: HttpClient, private router: Router) { }

  public create(priceList: PriceList) {
    return this.httpClient.post(this.url, priceList);
  }

  public getAll() {
    return this.httpClient.get(this.url);
  }

  public edit(priceList: PriceList) {
    return this.httpClient.put(this.url + "/" + priceList.id, priceList);
  }

  public delete(id: number) {
    return this.httpClient.delete(this.url + "/" + id);
  }

}

File: MainApp/AgentApp/Backend/src/main/java/jvn/RentACar/service/PriceListService.java
package jvn.RentACar.service;

import jvn.RentACar.model.PriceList;

import java.util.List;

public interface PriceListService {

    PriceList get(Long id);

    List<PriceList> getAll();

    PriceList create(PriceList priceList);

    PriceList edit(Long id, PriceList priceList);

    void delete(Long id);

}
